### PR TITLE
[BENCHMARK] dataset comparison igenvar only

### PR DESCRIPTION
**Pull request by Irallia**
UPDATE: look at the new plots below.

### In this plot you can see the results of iGenVar with 2 short read and 3 long read sets and their combinations.
iGenVar_only-results all
Some of the example BAM files (all HG002) are aligned to different references:
```
    MtSinai_PacBio:     GCA_000001405.15_GRCh38_no_alt_analysis_set.fa
    PacBio_CCS_10kb:    hs37d5.fa
    10X_Genomics:       hg19.reordered.fa
    Illumina:           GCA_000001405.15_GRCh38_no_alt_plus_hs38d1_analysis_set.fa
    Illumina_Mate_Pair: GCA_000001405.15_GRCh38_no_alt_plus_hs38d1_analysis_set.fa
```
### Since the truth set does not contain DUPs, I created another plot where all DUPs are interpreted as INS.
iGenVar_only-results DUP_as_INS all
### The important question now is, are most sets really that bad, or does iGenVar simply not find SVs. This I want to find out in comparison with other callers.

### CODE REVIEW

File: test/api/detection_test.cpp [hunk 1]
```diff
@@ -452,18 +452,18 @@ TEST(junction_detection, analyze_sa_tag)
...
         //      |         |   |        |   |                         |         |   |       |           |   |      |           |   |       |
-        // chr1 -1->      -2->-------> -3->------------>             -4->      -5->------->-6->        -7->------>-8->        -9->------->-10->
+        // 1    -1->      -2->-------> -3->------------>             -4->      -5->------->-6->        -7->------>-8->        -9->------->-10->
         //      ||||      |||||||||||| |||||||||||||||||\\\\\\\\\\\\ ||||      ||||||||||||||||        ||||       ||||        ||||
         // read --->-TRA->---><--INV-- --->-DUP:TANDEM->-DUP:TANDEM->--->-TRA->--->-DUP_1->--->--INS-->--->..DEL..--->-DUP_2->--->
         //          ||||||                                               ||||||                                       ||||||||
-        // chr2 --->----->--->                                  chr1 -9->----->-10->                         chr1 -5->------->-6->
+        // 2    --->----->--->                                  1 -9->----->-10->                         1 -5->------->-6->
```
Comment (Irallia) [inline test/api/detection_test.cpp:459]: some empty spaces are missing

File: test/api/detection_test.cpp [hunk 2]
```diff
@@ -238,7 +238,7 @@ TEST(junction_detection, analyze_aligned_segments)
     //
     //                  100   106       116           130                 150   156
     //                   |     |         |             |                   |     |
-    // chr1:             .....................................................................
+    // 1:                .....................................................................
```
Comment (Irallia) [inline test/api/detection_test.cpp:241]: maybe keep chr in comments

File: test/benchmark/caller_comparison_iGenVar_only/workflow/rules/eval.smk [hunk 3]
```diff
@@ -0,0 +1,150 @@
+# rule DUP_as_INS:
+#     input:
+#         vcf = "results/caller_comparison_iGenVar_only/{input_combination}/variants.vcf"
+#     output:
+#         vcf = "results/caller_comparison_iGenVar_only/{input_combination}/variants.DUP_as_INS.vcf"
+#     log:
+#         "logs/caller_comparison_iGenVar_only/DUP_as_INS.{input_combination}.log"
+#     shell:
+#         "sed -e 's/<DUP:TANDEM>/<INS>/g' {input.vcf} | sed -e 's/SVTYPE=DUP/SVTYPE=INS/g' > {output.vcf}"
```
Comment (Irallia) [inline test/benchmark/caller_comparison_iGenVar_only/workflow/rules/eval.smk:9]: remove

File: test/benchmark/caller_comparison_iGenVar_only/workflow/rules/eval.smk [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+rule filter_vcf:
+    input:
+        vcf = "results/caller_comparison_iGenVar_only/{input_combination}/variants.vcf",
+            #   "results/caller_comparison_iGenVar_only/{input_combination}/variants.DUP_as_INS.vcf"
+    output:
+        vcf = "results/caller_comparison_iGenVar_only/{input_combination}/variants.min_qual_{min_qual}.vcf",
+            #   "results/caller_comparison_iGenVar_only/{input_combination}/variants.DUP_as_INS.min_qual_{min_qual}.vcf"
+    log:
+        "logs/caller_comparison_iGenVar_only/filter_vcf_output.{input_combination}.{min_qual}.log",
+        # "logs/caller_comparison_iGenVar_only/filter_vcf_output.{input_combination}.DUP_as_INS.{min_qual}.log",
```
Comment (Irallia) [inline test/benchmark/caller_comparison_iGenVar_only/workflow/rules/eval.smk:20]: here aswell

File: test/benchmark/caller_comparison_iGenVar_only/workflow/scripts/plot_all_results.R [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+                               '0.1', '0.2', '0.3', '0.4', '0.5', '0.6', '0.7', '0.8', '0.9'),
+                      labels=c('Illumina', 'Illumina Mate Pair', 'Illumina & MtSinai PacBio', 'Illumina Mate Pair & MtSinai PacBio', 'Illumina & PacBio CCS', 'Illumina Mate Pair & PacBio CCS', 'Illumina & 10X Genomics', 'Illumina Mate Pair & 10X Genomics', 'MtSinai PacBio', 'PacBio CCS', '10X Genomics',
+                               '0.1', '0.2', '0.3', '0.4', '0.5', '0.6', '0.7', '0.8', '0.9'))
+
+scale_custom <- list(
+      # https://stackoverflow.com/questions/46803260/assigning-40-shapes-or-more-in-scale-shape-manual
+      scale_shape_manual(values = c(15, 15, 16, 16, 16, 16, 16, 16, 17, 17, 17, 46, 46, 46, 46, 46, 46, 46, 46, 46)),
+      # https://www.farb-tabelle.de/de/farbtabelle.htm
+      scale_color_manual(breaks = c('Illumina', 'Illumina Mate Pair', 'Illumina & MtSinai PacBio', 'Illumina Mate Pair & MtSinai PacBio', 'Illumina & PacBio CCS', 'Illumina Mate Pair & PacBio CCS', 'Illumina & 10X Genomics', 'Illumina Mate Pair & 10X Genomics', 'MtSinai PacBio', 'PacBio CCS', '10X Genomics'),
```
Comment (Irallia) [inline test/benchmark/caller_comparison_iGenVar_only/workflow/scripts/plot_all_results.R:33]:
```suggestion
      scale_color_manual(breaks = c('Illumina', 'Illumina Mate Pair', 
                                    'Illumina & MtSinai PacBio', 'Illumina Mate Pair & MtSinai PacBio', 'Illumina & PacBio CCS', 'Illumina Mate Pair & PacBio CCS', 'Illumina & 10X Genomics', 'Illumina Mate Pair & 10X Genomics', 
                                    'MtSinai PacBio', 'PacBio CCS', '10X Genomics'),
```

File: test/benchmark/caller_comparison_long_read/config/config.yaml [hunk 6]
```diff
@@ -2,8 +2,21 @@ long_bam: data/long_reads/HG002.Sequel.10kb.pbmm2.hs37d5.whatshap.haplotag.RTG.1
 long_md_bam: data/long_reads/HG002.Sequel.10kb.pbmm2.hs37d5.whatshap.haplotag.RTG.10x.trio.md.bam
 long_bai: data/long_reads/HG002.Sequel.10kb.pbmm2.hs37d5.whatshap.haplotag.RTG.10x.trio.bam.bai
```
Comment (Irallia) [inline test/benchmark/caller_comparison_long_read/config/config.yaml:4]: remove this file

File: test/benchmark/config/caller_comparison_config.yaml [hunk 7]
```diff
@@ -12,5 +49,5 @@ parameters:
 
 quality_ranges:
     from: 1
-    to: 80
+    to: 70
```
Comment (Irallia) [inline test/benchmark/config/caller_comparison_config.yaml:52]: does this make sense?

File: test/benchmark/envs/environment.yml [hunk 8]
```diff
@@ -5,7 +5,7 @@ channels:
   - bioconda
 dependencies:
   - mamba
-  - pbgzip
+  - bgzip # rule tabix - accepts only zipped files zipped with bgzip
```
Comment (Irallia) [inline test/benchmark/envs/environment.yml:8]: move one up

**Comment by Irallia** [comment]
Updated plots (GRCh37)
iGenVar_only-results all
DUPs as INS
iGenVar_only-results DUP_as_INS all

**Review by joergi-w — APPROVED** [review]

**Comment by joshuak94** [comment]
This is interesting! It makes sense that pacbio CCS reads are *much* easier to call SVs from: they have long read lengths and relatively high accuracy. However, it is a bit concerning that just illumina mate-pair reads result in such low accuracy. I'd be curious to see how this looks specifically with something like Deletions, since deletions can be detected somewhat robustly just via read-depth. 

Maybe if we can find out if there is a specific variant which is bringing the whole curve down, we can better understand what the issue there is.

**Review by joshuak94 — APPROVED** [review]

**Pull request merged**